src/Tests/Seam/When_dispatching_messages_in_receive_context.cs

```diff
@@ -58,16 +51,29 @@ public void SetUp()
             endpointOrientedTopology.Initialize(settings);
 
             // create the topologySectionManager
-            var topologyCreator = (ICreateTopologyInternal) container.Resolve(typeof(TopologyCreator));
-            sectionManager = container.Resolve<ITopologySectionManagerInternal>();
-            await topologyCreator.Create(sectionManager.DetermineResourcesToCreate(new QueueBindings()));
-            container.RegisterSingleton<TopologyOperator>();
+            var namespaceManagerCreator = new NamespaceManagerCreator(settings);
+
+            var namespaceManagerLifeCycleManagerInternal = new NamespaceManagerLifeCycleManagerInternal(namespaceManagerCreator);
+            var topologyCreator = new TopologyCreator(new AzureServiceBusSubscriptionCreatorV6(endpointOrientedTopology.Settings.SubscriptionSettings, settings), 
+                new AzureServiceBusQueueCreator(endpointOrientedTopology.Settings.QueueSettings,settings),  new AzureServiceBusTopicCreator(endpointOrientedTopology.Settings.TopicSettings), 
+                namespaceManagerLifeCycleManagerInternal);
+            var topologySectionManagerInternal = new EndpointOrientedTopologySectionManager(settings, container);
+            await topologyCreator.Create(topologySectionManagerInternal.DetermineResourcesToCreate(new QueueBindings()));
```

Comment:
In the core we do not enforce it in the tests. If you prefer to have it I'm happy